File: 8vim/src/main/java/inc/flide/vim8/structures/yaml/Part.java
package inc.flide.vim8.structures.yaml;

import com.fasterxml.jackson.annotation.JsonProperty;
import inc.flide.vim8.structures.Direction;
import java.util.List;
import java.util.Map;

public class Part {
    @JsonProperty(required = true)
    public Map<Direction, List<Action>> parts;
}

File: 8vim/src/main/kotlin/inc/flide/vim8/ime/layout/models/yaml/Part.kt
package inc.flide.vim8.ime.layout.models.yaml

import arrow.optics.optics
import com.fasterxml.jackson.annotation.JsonProperty
import inc.flide.vim8.ime.layout.models.Direction
import inc.flide.vim8.lib.ExcludeFromJacocoGeneratedReport

@ExcludeFromJacocoGeneratedReport
@optics
data class Part(@JsonProperty(required = true) val parts: Map<Direction, List<Action>> = mapOf()) {
    companion object
}
